src/Clerk.ts

```diff
@@ -169,17 +168,18 @@ export default class Clerk {
 
         Logger.debug(`sessionId from query: ${sessionId}`);
 
+        let session: (Session | undefined) = undefined;
         if (!sessionId) {
-          client = await this.clients.verifyClient(sessionToken);
-          sessionId = client.lastActiveSessionId;
+          const client = await this.clients.verifyClient(sessionToken);
+          session = client.sessions.find(session => session.id === client.lastActiveSessionId);
           Logger.debug(`lastActiveSessionId from client: ${sessionId}`);
+        } else {
+          session = await this.sessions.verifySession(
+              sessionId,
+              sessionToken
+          );
```

Comment:
🙃 Shall we also add a logger.debug statement here?

Reply:
Sure 👍